src/err.rs

```diff
@@ -38,7 +38,7 @@ impl Error for NlError {
 
 /// Serialization error
 #[derive(Debug)]
-pub struct SerError(String);
+pub struct SerError(pub String);
```

Comment:
I had to make this public in order to create a custom error for the new type of string serialization error. There are probably other solutions. But I wanted something simple for now to present my idea here.